DiscountDiscover/View Controllers/ProfileViewController.m

```diff
@@ -26,19 +28,78 @@ @implementation ProfileViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
    
-    PFUser *user = [PFUser currentUser];
-    self.nameLabel.text = user[@"name"];
-    self.usernameLabel.text = [@"@" stringByAppendingString:user.username];
-    self.emailLabel.text = user.email;
-    self.preferencesLabel.text = user[@"preferences"];
-    
-    if (user[@"profileImage"]) {
-        self.profileView = user[@"profileImage"];
+    self.user = [PFUser currentUser];
+    self.nameLabel.text = self.user[@"name"];
+    self.usernameLabel.text = [@"@" stringByAppendingString:self.user.username];
+    self.emailLabel.text = self.user.email;
+    self.preferencesLabel.text = self.user[@"preferences"];
+    
+    if (self.user[@"profileImage"]) {
+        self.profileView.file = self.user[@"profileImage"];
+        [self.profileView loadInBackground];
     }
     self.profileView.layer.cornerRadius = self.profileView.frame.size.width / 2;
     self.profileView.clipsToBounds = YES;
 }
 
+- (IBAction)takePicture:(id)sender {
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+    
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
```

Comment:
So if camera is available users can't use their existing pics?

Reply:
Usually in this case you would present a bottom sheet with the camera and library options if both are available.